src/distinguished_name.cpp

```diff
@@ -71,19 +77,24 @@ std::string _getEntryByNIDAsString(X509_NAME *x509, ASN1_NID nid)
 DistinguishedName DistinguishedName::fromX509Name(X509_NAME *x509)
 {
     Builder builder{};
-    builder.commonName(_getEntryByNIDAsString(x509, ASN1_NID::CommonName));
-    builder.countryName(_getEntryByNIDAsString(x509, ASN1_NID::CountryName));
-    builder.localityName(_getEntryByNIDAsString(x509, ASN1_NID::LocalityName));
-    builder.stateOrProvinceName(_getEntryByNIDAsString(x509, ASN1_NID::StateOrProvinceName));
-    builder.organizationalUnitName(_getEntryByNIDAsString(x509, ASN1_NID::OrganizationalUnitName));
-    builder.organizationName(_getEntryByNIDAsString(x509, ASN1_NID::OrganizationName));
-    builder.pkcs9EmailAddress(_getEntryByNIDAsString(x509, ASN1_NID::Pkcs9EmailAddress));
-    builder.serialNumber(_getEntryByNIDAsString(x509, ASN1_NID::SerialNumber));
-    builder.givenName(_getEntryByNIDAsString(x509, ASN1_NID::GivenName));
-    builder.userId(_getEntryByNIDAsString(x509, ASN1_NID::UserId));
+    builder.addAttributeByIndex(ASN1_NID::CommonName, x509);
```

Comment:
This whole process looks weird to me. We search for all of the attributes and then try to place them in the right location. What happens if there is an item in the Subject of a type that is not supported? It will end up in the list with an uninitialized SubjectAttribute object. That sounds potentially dangerous. It rather looks to me like the reasonable strategy here would be to iterate over the DN and add them 1:1 to the internal data vector.

Reply:
I dont see how I can iterate over the X509 attributes.

Reply:
This behaves like before. When the attribute cannot be found we leave it empty.